Fix broken dependencies on ServiceAccountCredentials

oc243 opened the pull request.

Commit:
Fix broken ServiceAccountCredentials dependency.

Fix for https://github.com/bgilbert/gcsannex/issues/5, using the fix
 recommended by @shawnsarwar.

Commit:
Remove httplib2 from have_google_api check.

httplib2 is not part of the google cloud storage API, but is included in the
check for it. This patch moves the import to outside of this check.

bgilbert commented:
Thanks for the PR.  There are a couple issues:
- As written, this change requires `GOOGLE_APPLICATION_CREDENTIALS` to be set on every invocation of gcsannex, rather than only during `initremote`.
- Since the oauth2client and httplib2 imports have been moved out of the try block, git-annex will no longer show a nice error message if either of those packages is unavailable.

oc243 commented:
Thanks for the review.

I'll look into point 1 to see if we can stop the env var being required.

Point 2 was an intentional change as if you don't have either  `oauth2client` and `httplib2` installed then I think that seeing `ImportError: No module named httplib2` is more useful than seeing `ImportError: Google Cloud client library not found` in knowing how to resolve this issue.

Commit:
Add support for oauth2client v4.

If you have oauth2client v4 installed you currently get this error:
```[2016-11-21 12:49:43.249612] git-annex-remote-gcs --> DEBUG file_cache is unavailable when using oauth2client >= 4.0.0
[2016-11-21 12:49:43.250053] file_cache is unavailable when using oauth2client >= 4.0.0
[2016-11-21 12:49:43.250322] git-annex-remote-gcs --> Traceback (most recent call last):
git-annex: external special remote protocol error, unexpectedly received "Traceback (most recent call last):" (unable to parse command)
```

The reason is that
https://github.com/google/google-api-python-client/blob/master/googleapiclient/discovery_cache/file_cache.py
raises an ImportError if oauth2client is v4 or greater. However, googleapiclient
does indeed support oauth2client.

This patch prevents gcsannex from using the file cache if oauth2client is v4 or
greater.

oc243 commented:
Hi @bgilbert,

Just came across an example showing why I think that removing the `have_google_api` check is useful. On a new machine I got this error:
```
git annex enableremote google
enableremote google git-annex: external special remote error: Google Cloud client library not found
```
To debug it I removed the `have_google_api` logic and python then crashed early with what I think it a more useful error:
```
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin/git-annex-remote-gcs", line 11, in <module>
    load_entry_point('gcsannex==0.2', 'console_scripts', 'git-annex-remote-gcs')()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pkg_resources/__init__.py", line 560, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2648, in load_entry_point
    return ep.load()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2302, in load
    return self.resolve()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2308, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "build/bdist.macosx-10.12-x86_64/egg/gcsannex.py", line 39, in <module>
ImportError: cannot import name SignedJwtAssertionCredentials
git-annex: external special remote protocol error, unexpectedly received "" (unable to parse command)
CallStack (from HasCallStack):
  error, called at ./Remote/External.hs:335:34 in main:Remote.External
```
I now know that I need to install `SignedJwtAssertionCredentials`

Commit:
Fix gcsannex with googleapiclient v1.6.

googleapiclient v1.6 only allows you to pass one of `credentials` and `http` to
the discovery `build` functions [0]. Therefore the current version of gcsannex gives
this when you run `enableremote`:

```
enableremote google git-annex: Arguments http and credentials are mutually exclusive.
CallStack (from HasCallStack):
 error, called at ./Remote/External.hs:127:69 in main:Remote.External
```

This does remove the TIMEOUT paremeter. I can't see a way of using that
parameter in the new version of googleapiclient.

[0] https://github.com/google/google-api-python-client/blob/master/googleapiclient/discovery.py#L337
